### ISSUE TITLE
categorical_crossentropy fails when using variable size input in the Fully Convolutional Network case

### ISSUE THREAD

**Issue by lukovkin** [fchollet/keras#2411]
After upgrading to Keras 1.0.1 and migrating of the model to Functional API, failed to use categorical_crossentropy as loss function.

Gives following error both on Theano and TensorFlow:

> Exception: A target array with shape (1, 50000, 3) was passed for an output of shape (None, None, 3) while using as loss `categorical_crossentropy`. This loss expects targets to have the same shape as the output.

On Keras 0.3.3 Graph API model was working fine with categorical_crossentropy.

Code - https://github.com/lukovkin/ufcnn-keras/blob/conv_transpose/models/UFCNN_functional.py

Model initialization and training after line 1211.

Please make sure that the boxes below are checked before you submit your issue. Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by NasenSpray** [comment]
Comment out this block https://github.com/fchollet/keras/blob/b61235b77f87288d62ddd8ce4aae88b76babf887/keras/engine/training.py#L198-L204

**Comment by lukovkin** [comment]
@NasenSpray Thank you! Tried it, works, it's OK as a workaround.
But I think it should be fixed somehow in the future.

**Comment by fchollet** [comment]
I have pushed a fix. Thanks for the report.
